GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -161,6 +157,24 @@ protected override void Dispose(bool disposing)
             base.Dispose(disposing);
         }
 
+        protected override void OnUICommandsSourceSet(IGitUICommandsSource source)
+        {
+            base.OnUICommandsSourceSet(source);
+
+            if (source is null)
+            {
+                _linkFactory = null;
+                _commitDataBodyRenderer = null;
+                _refsFormatter = null;
+            }
+            else
+            {
+                _linkFactory = source.UICommands.GetRequiredService<ILinkFactory>();
+                _commitDataBodyRenderer = new CommitDataBodyRenderer(() => Module, _linkFactory);
+                _refsFormatter = new RefsFormatter(_linkFactory);
+            }
+        }
+
```

Comment:
Why have you discarded 0a02a1b2e9354793f22cbf3dae11834c5e5984aa?

Reply:
I likely lost it during the rebase, fixup and cherrypicks